llvm/include/llvm/SandboxIR/PassManager.h

```diff
@@ -0,0 +1,74 @@
+//===- PassManager.h --------------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Registers and executes the Sandbox IR passes.
+//
+// The pass manager contains an ordered sequence of passes that it runs in
```

Comment:
it seems error prone for the pass manager to not own the pass (e.g. stack references). e.g. in the new pass manager if you want to run a pass multiple times, you just create multiple instances of it

Reply:
The pass registry would own the passes and would also provide the name-to-pass lookup so that you can build the pipeline from a pipeline string like "pass1,pass2".

If the pass manager owns the passes, then how is the mapping from names to passes handled?

Reply:
see includes of `PassRegistry.def`, especially in `PassBuilder.cpp`. it's basically a string comparison chain

I think your way is fine, as long as your explanation of the registry owning passes is documented

Reply:
I see, it shouldn't be too hard to follow a similar design. Let me try it out.

Reply:
I think I will stick with this design for now. Using a PassRegistry.def complicates the design quite a bit and I am not sure it is worth it at this point. We can revisit it later if needed.